qemu/tests/fio_windows.py

```diff
@@ -0,0 +1,101 @@
+import re
+import time
+import logging
+
+from autotest.client.shared import error
+from autotest.client.shared import data_dir
+from autotest.client import utils
+
+from virttest import utils_misc
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    KVM guest stop test:
+    1) Log into a guest
+    2) Check is fio.msi installed , download and
+       install it if not installed.
+    3) Start fio in guest
+    4) Check vm is alive
+    5) Stop fio process and clean temp file.
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def is_cmd_finished(session, cmd):
+        """
+        Return the status after doing cmd.
+        """
+        return session.cmd_status(cmd, timeout=120) == 0
+
+    tmp_dir = data_dir.get_tmp_dir()
+    install_path = params["install_path"].rstrip("\\")
+    start_cmd = params.get("start_cmd")
+    stop_cmd = "taskkill /T /F /IM fio.exe"
+    stop_cmd = params.get("stop_cmd", stop_cmd)
+    check_running_cmd = "tasklist|findstr /I fio"
+    check_running_cmd = params.get("check_running_cmd", check_running_cmd)
+    check_installed_cmd = 'dir "%s"|findstr /I fio' % install_path
+    check_installed_cmd = params.get("check_installed_cmd",
+                                     check_installed_cmd)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+    # Start another session with the guest and
+    # make sure the background process is running
+    session2 = vm.wait_for_login(timeout=timeout)
+    try:
+        installed = session.cmd_status(check_installed_cmd) == 0
+        if not installed:
+            download_url = params.get("download_url")
+            if download_url:
+                dst = r"c:\\"
```

Comment:
we have winutils to save windows guest test related tools, so only support installed fio from winutils is enough for us .